code/modules/client/viewport.dm

```diff
@@ -0,0 +1,85 @@
+GLOBAL_VAR_INIT(lock_client_view_x, null)
+GLOBAL_VAR_INIT(lock_client_view_y, null)
+GLOBAL_LIST_INIT(valid_icon_sizes, list(32, 48, 64, 72, 96, 128))
+
+/datum/preferences
+	var/icon_size = 64
+
+/client
+	var/last_view_x_dim = 7
+	var/last_view_y_dim = 7
```

Comment:
I'd prefer these to not be isolated from the rest of the pref & client var defines THOUGH this is fine for now as we still need to go through with our preference refactor